Stripe/STPSourcePoller.m

```diff
@@ -0,0 +1,108 @@
+//
+//  STPSourcePoller.m
+//  Stripe
+//
+//  Created by Ben Guo on 1/26/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import "STPAPIClient+Private.h"
+#import "STPAPIRequest.h"
+#import "STPSource.h"
+#import "STPSourcePoller.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+static NSTimeInterval const DefaultPollInterval = 1.5;
+static NSTimeInterval const MaxPollInterval = 24;
+
+@interface STPSourcePoller ()
+
+@property (nonatomic, weak) STPAPIClient *apiClient;
+@property (nonatomic) NSString *sourceID;
+@property (nonatomic) NSString *clientSecret;
+@property (nonatomic, copy) STPSourceCompletionBlock completion;
+@property (nonatomic, nullable) STPSource *latestSource;
+@property (nonatomic) NSTimeInterval pollInterval;
+@property (nonatomic) BOOL shouldStopPolling;
+
+@end
+
+@implementation STPSourcePoller
+
+- (instancetype)initWithAPIClient:(STPAPIClient *)apiClient
+                     clientSecret:(NSString *)clientSecret
+                         sourceID:(NSString *)sourceID
+                       completion:(STPSourceCompletionBlock)completion {
+    self = [super init];
+    if (self) {
+        _apiClient = apiClient;
+        _sourceID = sourceID;
+        _clientSecret = clientSecret;
+        _completion = completion;
+        _pollInterval = DefaultPollInterval;
+        _shouldStopPolling = NO;
+        [self pollAfter:0];
+    }
+    return self;
+}
+
+- (void)pollAfter:(NSTimeInterval)interval {
+    if (!self.apiClient) {
+        [self stopPolling];
+        return;
+    }
+    dispatch_time_t time = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(interval * NSEC_PER_SEC));
```

Comment:
Would it be better to use NSTimer here so we can stop immediately when a stopPolling is called? You can't cancel a queued dispatch.